### PR TITLE
[Swift] Remove all the hardcoded occurrences of mangling prefixes.

### PR DESCRIPTION
**Pull request by dcci**
<rdar://problem/44467875>

### CODE REVIEW

**Comment by dcci** [comment]
@swift-ci test and merge

**Comment by dcci** [comment]
I wasn't entirely sure whether the correct spelling was `prefixes` or `prefices`, but I'm under the impression nobody really cares as long as it's understandable.

**Comment by jimingham** [comment]
There's some contention about `prefices`.  The consensus is that prefices isn't the right pluralization of prefix, since "prefix" isn't actually a latin word.  Instead it is "derived from praefixum" and it's plural is praefices...  Since it isn't a direct latin word, using the latinate pluralization is considered wrong.  OTOH, the OED does cite `prefices` as a sometimes used alternate form of `prefixes`.

But the patch looks fine.  I wonder if in the last diff we're providing too much information.  We don't actually know that the only check IsSwiftMangledName does (or will do in the future) is the prefix, so by saying the prefix is the problem, we're speaking past what we know.  Maybe it would be better for that one to just say it isn't a valid swift name.

**Comment by dcci** [comment]
I agree the message could be improved. In general, this is fundamentally something people shouldn't care about (unless they're developing lldb proper).

The error is returned by `GetTypeFromMangledTypename()` which is an internal function of lldb. It should be the caller responsibility to filter out invalid names, even though being defensive here doesn't hurt.

File: source/Symbol/SwiftASTContext.cpp [hunk 1]
```diff
@@ -3862,9 +3863,10 @@ SwiftASTContext::GetTypeFromMangledTypename(const char *mangled_typename,
 
   if (!mangled_typename ||
       !SwiftLanguageRuntime::IsSwiftMangledName(mangled_typename)) {
-    error.SetErrorStringWithFormat("typename '%s' is not a valid Swift mangled "
-                                   "typename, it should begin with $S",
-                                   mangled_typename);
+    error.SetErrorStringWithFormat(
+        "typename '%s' is not a valid Swift mangled "
+        "typename, it should begin with " MANGLING_PREFIX_STR,
+        mangled_typename);
```
Comment (adrian-prantl) [inline source/Symbol/SwiftASTContext.cpp:3869]: This error message is way too specific. It should just say ... is not a valid Swift mangled name. No consumer will read this error and think "Oh, I need to change the prefix!" :-)

File: source/Plugins/Language/Swift/SwiftFormatters.cpp [hunk 2]
```diff
@@ -486,8 +487,7 @@ bool lldb_private::formatters::swift::NSContiguousString_SummaryProvider(
   if (guts_sp)
     return StringGuts_SummaryProvider(*guts_sp, stream, options);
 
-  static ConstString g_StringGutsType(
-      SwiftLanguageRuntime::GetCurrentMangledName("$Ss11_StringGutsVD"));
+  static ConstString g_StringGutsType(MANGLING_PREFIX_STR "s11_StringGutsVD");
```
Comment (adrian-prantl) [inline source/Plugins/Language/Swift/SwiftFormatters.cpp:490]: This looks dangerous. This should be code that tree-matches the demangle tree instead.

Comment (dcci) [reply source/Plugins/Language/Swift/SwiftFormatters.cpp:490]: I mean, sure, but it's worse than any other formatter in-tree? I don't think it's a problem specific to this very instance.

File: source/Symbol/SwiftASTContext.cpp (hunk 1 above)
Comment (dcci) [reply source/Symbol/SwiftASTContext.cpp:3869]: Agreed, see previous comment :)

**Pull request merged**